flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java

```diff
@@ -193,22 +193,50 @@ protected File getPackageLockFile() {
      *             when versions file could not be read
      */
     JsonObject getPlatformPinnedDependencies() throws IOException {
-        URL resource = finder.getResource(Constants.VAADIN_VERSIONS_JSON);
-        if (resource == null) {
-            log().info("Couldn't find {} file to pin dependency versions."
-                    + " Transitive dependencies won't be pinned for pnpm.",
-                    Constants.VAADIN_VERSIONS_JSON);
+        URL coreVersionsResource = finder
+                .getResource(Constants.VAADIN_CORE_VERSIONS_JSON);
+        if (coreVersionsResource == null) {
+            log().info(
+                    "Couldn't find {} file to pin dependency versions for core components."
+                            + " Transitive dependencies won't be pinned for npm/pnpm.",
+                    Constants.VAADIN_CORE_VERSIONS_JSON);
             return Json.createObject();
         }
 
         JsonObject versionsJson;
-        try (InputStream content = resource.openStream()) {
+        try (InputStream content = coreVersionsResource.openStream()) {
             VersionsJsonConverter convert = new VersionsJsonConverter(Json
                     .parse(IOUtils.toString(content, StandardCharsets.UTF_8)));
             versionsJson = convert.getConvertedJson();
             versionsJson = new VersionsJsonFilter(getPackageJson(),
-                    DEPENDENCIES).getFilteredVersions(versionsJson);
+                    DEPENDENCIES).getFilteredVersions(versionsJson,
+                            Constants.VAADIN_CORE_VERSIONS_JSON);
+        }
+
+        URL commVersionsResource = finder
```

Comment:
```suggestion
        URL vaadinVersionsResource = finder
```
commVersionsResource sounds like some communication package or other. Would call it vaadinVersionsResource or just versionsResource (to be the same as core) to reflect better what it is.

Reply:
Done.